apsuite/commisslib/meas_bo_tune_by_bpm.py

```diff
@@ -105,41 +103,49 @@ def get_orbit_data(self, injection=False, external_trigger=False):
             self.devices['evg'].cmd_turn_on_injection()
         ret = bobpms.mturn_wait_update_flags(timeout=prms.bpms_timeout)
         if ret:
-            trigbpm.delay = self.delay0
+            trigbpm.delay = delay0
             self.data = dict()
-            raise AssertionError(f'Problem waiting BPMs update. Error code: {ret:d}')
+            raise AssertionError(
+                f'Problem waiting BPMs update. Error code: {ret:d}')
         orbx, orby = bobpms.get_mturn_orbit()
         bobpms.cmd_mturn_acq_abort()
-        trigbpm.delay = self.delay0
+        trigbpm.delay = delay0
+
+        self.data['orbx'], self.data['orby'] = orbx, orby
+        self.data['timestamp'] = _time.time()
+
+    def get_data(self, delta=''):
+        """."""
 
         # Store data
+        bobpms = self.devices['bobpms']
+        trigbpm = self.devices['trigbpm']
+
         bpm0 = bobpms[0]
         csbpm = bpm0.csdata
         data = dict()
-        data['timestamp'] = _time.time()
+        data['delta'] = delta
         data['rf_frequency'] = self.devices['rfgen'].frequency
         data['current_150mev'] = self.devices['currinfo'].current150mev
         data['current_1gev'] = self.devices['currinfo'].current1gev
         data['current_2gev'] = self.devices['currinfo'].current2gev
         data['current_3gev'] = self.devices['currinfo'].current3gev
-        data['orbx'], data['orby'] = orbx, orby
         data['mt_acq_rate'] = csbpm.AcqChan._fields[bpm0.acq_channel]
         data['bpms_nrsamples_pre'] = bpm0.acq_nrsamples_pre
         data['bpms_nrsamples_post'] = bpm0.acq_nrsamples_post
         data['bpms_trig_delay_raw'] = trigbpm.delay_raw
         data['bpms_switching_mode'] = csbpm.SwModes._fields[
                                         bpm0.switching_mode]
-        data['init_delay'] = self.delay0
-        self.data = data
+        data['qf_wave_form'] = PV('BO-Fam:PS-QF:Wfm-RB').get()
```

Comment:
This is not a good practice. You should create the PVs you need at the object initialization, together with the devices creation.  Add the PVs to the dictionary `self.pvs` which is already available for this type of object. All PVs added to this dictionary will be checked when the method `self.connected` or `self.wait_for_connection` is called.

Reply:
Ok, done!